src/wallet/rpcdump.cpp

```diff
@@ -572,7 +576,9 @@ UniValue dumpwallet(const JSONRPCRequest& request)
     EnsureWalletIsUnlocked();
 
     ofstream file;
-    file.open(request.params[0].get_str().c_str());
+    boost::filesystem::path filepath = request.params[0].get_str();
+    filepath = absolute(filepath);
```

Comment:
Interesting. Shouldn't it be only accepting absolute paths in the first place?
Otherwise it will depend on the directory that bitcoind happens to have been started in, that'd be pretty crappy API design.